diff -Naur surf-0.7/config.h surf-0.7_copy/config.h
--- surf-0.7/config.h	2016-12-27 13:16:28.326644513 +0100
+++ surf-0.7_copy/config.h	2016-12-27 13:14:50.122646529 +0100
@@ -37,10 +37,9 @@
 
 #define SETPROP(p, q) { \
 	.v = (char *[]){ "/bin/sh", "-c", \
-	     "prop=\"`xprop -id $2 $0 " \
-	     "| sed \"s/^$0(STRING) = \\(\\\\\"\\?\\)\\(.*\\)\\1$/\\2/\" " \
-	     "| xargs -0 printf %b | dmenu`\" &&" \
-	     "xprop -id $2 -f $1 8s -set $1 \"$prop\"", \
+                   "prop=\"`(xprop -id $2 $0 | cut -d '\"' -f 2 | xargs -0 printf %b && " \
+                   "cat ~/.surf/bookmarks) | dmenu`\" &&" \
+	     	     "xprop -id $2 -f $1 8s -set $1 \"$prop\"", \
 	     p, q, winid, NULL \
 	} \
 }
@@ -54,6 +53,11 @@
 	     d, useragent, r, cookiefile, NULL \
 	} \
 }
+#define BM_ADD { .v = (char *[]){ "/bin/sh", "-c",                      \
+                                    "(echo `xprop -id $0 _SURF_URI | cut -d '\"' -f 2` && " \
+                                    "cat ~/.surf/bookmarks) | awk '!seen[$0]++' > ~/.surf/bookmarks_new && " \
+                                    "mv ~/.surf/bookmarks_new ~/.surf/bookmarks", \
+                                    winid, NULL } }
 
 /* PLUMB(URI) */
 /* This called when some URI which does not begin with "about:",
@@ -107,6 +111,7 @@
 	{ MODKEY,               GDK_i,      scroll_h,   { .i = +1 } },
 	{ MODKEY,               GDK_u,      scroll_h,   { .i = -1 } },
 
+    { MODKEY,               GDK_b,      spawn,      BM_ADD },
 	{ 0,                    GDK_F11,    fullscreen, { 0 } },
 	{ 0,                    GDK_Escape, stop,       { 0 } },
 	{ MODKEY,               GDK_o,      source,     { 0 } },
