diff -Naur surf-0.7/config.h surf-0.7_copy/config.h
--- surf-0.7/config.h	2016-12-27 13:16:28.326644513 +0100
+++ surf-0.7_copy/config.h	2016-12-27 13:23:41.222635630 +0100
@@ -97,23 +97,26 @@
 	{ MODKEY,               GDK_minus,  zoom,       { .i = -1 } },
 	{ MODKEY,               GDK_plus,   zoom,       { .i = +1 } },
 
-	{ MODKEY,               GDK_l,      navigate,   { .i = +1 } },
-	{ MODKEY,               GDK_h,      navigate,   { .i = -1 } },
+    { MODKEY,               GDK_i,      navigate,   { .i = +1 } },
+    { MODKEY,               GDK_o,      navigate,   { .i = -1 } },
+
 
 	{ MODKEY,               GDK_j,      scroll_v,   { .i = +1 } },
 	{ MODKEY,               GDK_k,      scroll_v,   { .i = -1 } },
 	{ MODKEY,               GDK_b,      scroll_v,   { .i = -10000 } },
+    { MODKEY,               GDK_f,      scroll_v,   { .i = +10000 } },
 	{ MODKEY,               GDK_space,  scroll_v,   { .i = +10000 } },
-	{ MODKEY,               GDK_i,      scroll_h,   { .i = +1 } },
-	{ MODKEY,               GDK_u,      scroll_h,   { .i = -1 } },
+    { MODKEY,               GDK_l,      scroll_h,   { .i = +1 } },
+    { MODKEY,               GDK_h,      scroll_h,   { .i = -1 } },
+
 
 	{ 0,                    GDK_F11,    fullscreen, { 0 } },
 	{ 0,                    GDK_Escape, stop,       { 0 } },
-	{ MODKEY,               GDK_o,      source,     { 0 } },
-	{ MODKEY|GDK_SHIFT_MASK,GDK_o,      inspector,  { 0 } },
+    { MODKEY,               GDK_u,      source,     { 0 } },
+    { MODKEY|GDK_SHIFT_MASK,GDK_u,      inspector,  { 0 } },
+
 
 	{ MODKEY,               GDK_g,      spawn,      SETPROP("_SURF_URI", "_SURF_GO") },
-	{ MODKEY,               GDK_f,      spawn,      SETPROP("_SURF_FIND", "_SURF_FIND") },
 	{ MODKEY,               GDK_slash,  spawn,      SETPROP("_SURF_FIND", "_SURF_FIND") },
 
 	{ MODKEY,               GDK_n,      find,       { .b = TRUE } },
